constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,133 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterRewriteDataRef(t *testing.T) {
+	tests := []struct {
+		name    string
+		prefix  string
+		imports string
+		want    string
+	}{
+		{
+			name:    "no rewrite for ref is sub of externs",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			want:    "data.ext",
+		},
+		{
+			name:    "no rewrite for non data ref",
+			prefix:  "foo.bar",
+			imports: "invalid_data.name",
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.lib.util",
+			want:    "data.data.lib.lib.util",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, err := New(NewPackagePrefixer(tt.prefix), []string{"data.lib"}, []string{"data.ext", "data.inventory"})
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			ref, err := ast.ParseRef(tt.imports)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			wantRef, err := ast.ParseRef(tt.want)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			if got := r.rewriteDataRef(ref); !got.Equal(wantRef) {
+				t.Errorf("rewriteDataRef() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestAddEntryPointParseErrors(t *testing.T) {
+	tcs := []struct {
+		name      string
+		path      string
+		src       string
+		wantError error
+	}{
+		{
+			name: "add lib.rego path",
+			path: "data.stuff.foolib",
+			src: RegoSrc("lib.rego", Body(`test_ok {
+						true
+					}`)),
+		},
+		{
+			name:      "add invalid rego",
+			path:      "invalidrego",
+			src:       "something invalid",
+			wantError: ErrInvalidModule,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := MockRegoWriter()
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			if gotErr := rr.AddEntryPoint(tc.path, tc.src); !errors.Is(gotErr, tc.wantError) {
+				t.Errorf("got AddEntryPoint() %s error = %v, want %v", tc.name, gotErr, tc.wantError)
```

Comment:
Don't put `tc.name` in failure messages. Since we're using `t.Error` and similar, the test case name is already in the failure block.

Reply:
Ack. Removed tc.name, it was handy for searching failed tests.